agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/FileStringLookup.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.agent.internal.configuration;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Paths;
+import javax.annotation.Nullable;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.text.lookup.StringLookup;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class FileStringLookup implements StringLookup {
+
+  private static final Logger logger = LoggerFactory.getLogger(FileStringLookup.class);
+  static final FileStringLookup INSTANCE = new FileStringLookup();
+
+  @Override
+  @Nullable
+  public String lookup(final String key) {
+    if (key == null) {
+      return null;
+    }
+
+    final String fileName = StringUtils.substringAfter(key, ":");
+    try {
+      return new String(Files.readAllBytes(Paths.get(fileName)), StandardCharsets.UTF_8);
+    } catch (IOException | InvalidPathException e) {
+      logger.error(
+          "Error occurs when reading connection string from the file '{}' with UTF-8 encoding.",
+          fileName,
+          e);
+      return null;
```

Comment:
I think rethrow here to fail agent startup with this message

Reply:
rethrow will missing the env var override... i will let setConnectionString handle it.. it will through exception a bit down.

Reply:
existing behavior is to fail on any json configuration error, and not try to recover or handle overrides in that case, I think good to stay consistent with the this current fail-fast behavior

Reply:
i see. will do.